contracts/competition/Competition.sol

```diff
@@ -0,0 +1,138 @@
+pragma solidity ^0.6.0;
+
+import "../referrals/interfaces/IReferralStorage.sol";
+
+contract Competition
+{
+    uint public start;
+    uint public end;
+    uint public registrationStart;
+    uint public registrationEnd;
+    IReferralStorage private referralStorage;
+
+    struct Team {
+        address leader;
+        string name;
+        bytes32 referral;
+        address[] members;
+        address[] joinRequests;
+    }
+
+    address[] private leaders;
```

Comment:
mm why do we need this array? do we want to show list of all readers somewhere in UI?

Reply:
Will be used to fetch teams addresses (leader address) for the leaderboard
We will probably end up using a graph so it's just an option right now

Reply:
i see. yeah, i was thinking about subgraph as well

at the same time registration happens once and probably it's okay to store it in contract 🤔